src/fmt/mod.rs

```diff
@@ -289,6 +294,28 @@ impl<'a> DefaultFormat<'a> {
     fn write_args(&mut self, record: &Record) -> io::Result<()> {
         writeln!(self.buf, "{}", record.args())
     }
+
+    fn write_kv(&mut self, record: &Record) -> io::Result<()> {
+        if !self.key_values {
+            return Ok(())
+        }
+        let kvs = record.key_values();
+        if !self.written_header_value && kvs.count() > 0 {
```

Comment:
I don't love this, but I couldn't think of a way of doing it I liked any better.